src/main.rs

```diff
@@ -535,92 +550,77 @@ impl ManParser for Type1 {
         manpage.contains(r#".SH "OPTIONS""#)
     }
 
-    fn parse_man_page(&mut self, manpage: &str) -> Option<String> {
-        unimplemented!();
+    fn parse_man_page(&self, manpage: &str, cmdname: &str) -> Option<String> {
+        let options_section_re = regex!(r#"\.SH "OPTIONS"((?s:.)*?)(\.SH|\Z)"#);
```

Comment:
The `Type2` equivalent regex also uses `\Z`, but the `Type3` and `Type4` equivalents use `\\Z`. Is that intentional?

If possible, would be good to use the same style of `r#"..."#` or `r"..."` for all 4 so that the intentional differences between the regexes are more obvious to the reader.

Reply:
Ops, `\Z` is the wrong one, I fixed it.
```suggestion
        let options_section_re = regex!(r#"\.SH "OPTIONS"((?s:.)*?)(\.SH|\\Z)"#);
```

Reply:
I do want to keep consistency between `r#"..."#` or `r"..."`. But the main reason `r#"..."#` was used is because there is `"` inside.

Reply:
Ah, good point. I missed that. No need to change the quote style.